src/main/java/com/minecolonies/coremod/colony/managers/ResearchManager.java

```diff
@@ -45,4 +80,123 @@ public IResearchEffectManager getResearchEffects()
     {
         return this.effects;
     }
+
+    @Override
+    public void onColonyTick(final IColony colony)
+    {
+        checkAutoStartResearch(colony);
+        checkInprogressResearch(colony);
+    }
+
+    /**
+     * Checks if any autostart research has its prerequisites filled,
+     * and if so, prompts the player for resources or begins research if no resources required.
+     * @param colony       The colony where the research is being checked.
+     */
+    private void checkAutoStartResearch(final IColony colony)
```

Comment:
I'm not fully sure I grasp the idea behind Auto start research. What are the use cases for it? Don't we need a university for this too?

Reply:
I think Mekle001’s central use case involved situations where the university itself was locked, such as wanting to make it unbuildable until after a Town Hall T1 was constructed. Without having access to the WindowResearchTree GUI, you’d need some way to start it. 

Quest-focused modpacks can kinda handle things by removing the University hut block recipe and making it a quest reward, but that’s vulnerable to problems (ie, losing it to lava).

Or if a modpack just wants to make a very linear path through the huts, it would save walking to the Uni and back each time for 10-20 unlocks. 

The other situation, and one more common for your use, would just be to notify on reveal of Hidden research.  If More Scrolls should only be visible after getting a T3 Enchanter, it makes sense to tell the player that the new research is there — but it may not be desirable to always do such notification.

Reply:
I get the notification part. But for that we don't need to auto start the research. We'd just need an event based approach and a "on unlock message" that may be defined for a research. This way no ticking is required.

For the "lock/unlock" things without a university I'd probably go the same way. Just an unlock event